.github/workflows/ci.yml

```diff
@@ -0,0 +1,172 @@
+    - name: Install dependencies
+      run: yarn --immutable
+
+    - name: Lint code
+      run: yarn test:lint
+
+    - name: Run Flow
+      run: yarn test:flow
+      if: always()
+
+    - name: Run Jest tests
+      run: yarn test:jest --coverage
+      if: always()
+
+    - uses: codecov/codecov-action@v1
+      with:
+        directory: ./coverage/
+
+  screenshotter_dispatcher:
+    runs-on: ubuntu-latest
+    if: |
+      (github.event_name != 'pull_request_target' ||
+        (github.event.pull_request.head.repo.full_name != 'KaTeX/KaTeX' &&
+          contains(github.event.pull_request.labels.*.name, 'test screenshots'))) &&
+      !contains(toJSON(github.event.commits.*.message), '[skip ci]') &&
+      !contains(toJSON(github.event.commits.*.message), '[ci skip]')
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.result }}
+    
+    steps:
+    - id: set-matrix
+      uses: actions/github-script@v2
+      with:
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+        script: |
+          const SELENIUM_BROWSERS = ["chrome", "firefox"];
+          const BROWSERSTACK_BROWSERS = [{
+            browserName: "safari",
+            browser_version: "13.1",
+            os: "OS X",
+            os_version: "Catalina",
+          }];
+
+          const include = [];
+
+          if (context.eventName !== "pull_request_target") {
+            include.push(...SELENIUM_BROWSERS.map(b => ({
+              browser: b,
+              services: {selenium: {
+                image: `selenium/standalone-${b}:3.141.59-20200525`,
+                ports: ["4444:4444"],
+              }},
+            })));
+          }
+
+          // check access to Browserstack crendential secrets
+          if (context.eventName !== "pull_request" ||
+              context.payload.pull_request.head.repo.full_name === "KaTeX/KaTeX") {
+            if (context.eventName === "pull_request_target") {
+              github.issues.removeLabel({
+                issue_number: context.issue.number,
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                name: "test screenshots",
+              });
+            }
+            include.push(...BROWSERSTACK_BROWSERS.map(b => ({
+              browser: b.browserName,
+              services: {},
+              browserstack: b,
+            })));
+          }
+
+          return {browser: include.map(b => b.browser), include};
+
+  screenshotter:
+    runs-on: ubuntu-latest
+    needs: screenshotter_dispatcher
+    strategy:
+      matrix: ${{ fromJson(needs.screenshotter_dispatcher.outputs.matrix) }}
+      fail-fast: false
+    services: ${{ matrix.services }}
+
+    steps:
+    - uses: actions/checkout@v2
+      if: github.event_name != 'pull_request_target'
+      with:
+        submodules: recursive
+    - uses: actions/checkout@v2
+      if: github.event_name == 'pull_request_target'
+      with:
+        ref: refs/pull/${{ github.event.pull_request.number }}/head
```

Comment:
Why do we have to specify the ref when using `pull_request_target`?

Reply:
@kevinbarabash `pull_request_target` is run [in the context of the base repository of the pull request](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target), so the default ref is `master` branch, not the head of the PR.

Reply:
That makes sense.  Could you add a quick comment here saying that?